### PR TITLE
update: merging curl tutorial with the native hands on

### PR DESCRIPTION
**Pull request by danil-pavlov**

### CODE REVIEW

File: docs/topics/native/native-app-with-c-and-libcurl.md [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+
+To get started, install the latest version of [IntelliJ IDEA](https://www.jetbrains.com/idea/download/index.html).
+The tutorial is applicable to both IntelliJ IDEA Community Edition and the Ultimate Edition.
+
+## Create a Kotlin/Native project
+
+1. In IntelliJ IDEA, select **File | New | Project**.
+2. In the panel on the left, select **Kotlin | Native Application**.
+3. Specify the name and select the folder where you'll save your application.
+   ![New project. Native application in IntelliJ IDEA](native-file-new.png){width="700"}
```
Comment (koshachy) [inline docs/topics/native/native-app-with-c-and-libcurl.md:24]: You can omit quotes for the width number. `width=700`

File: docs/topics/native/native-app-with-c-and-libcurl.md [hunk 2]
```diff
@@ -0,0 +1,260 @@
+[//]: # (title: Create an app using C Interop and libcurl – tutorial)
+
+This tutorial demonstrates how to use IntelliJ IDEA for creating a command-line application. You'll learn how to create
+a simple HTTP client that can run natively on specified platforms using Kotlin/Native and the `libcurl` library.
+
+The output will be an executable command-line app that you can run on macOS and Linux and make simple HTTP GET requests.
```
Comment (koshachy) [inline docs/topics/native/native-app-with-c-and-libcurl.md:6]: Does it seem that we need a comma here, right after Linux?

File: docs/topics/native/native-app-with-c-and-libcurl.md [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+```c
+headers = curl/curl.h
+headerFilter = curl/*
+
+compilerOpts.linux = -I/usr/include -I/usr/include/x86_64-linux-gnu
+linkerOpts.osx = -L/opt/local/lib -L/usr/local/opt/curl/lib -lcurl
+linkerOpts.linux = -L/usr/lib/x86_64-linux-gnu -lcurl
+```
+
+* The first entry is `headers`, the list of header files to generate Kotlin stubs for. You can add multiple files to
```
Comment (koshachy) [inline docs/topics/native/native-app-with-c-and-libcurl.md:123]: This easily can be simplified to "`headers` contains the list of header files ...".
But the current version is also OK.

File: docs/topics/native/native-app-with-c-and-libcurl.md [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+this entry, separating each with a `\` on a new line. In our case, it's only `curl.h`. The referenced files need to be
```
Comment (koshachy) [inline docs/topics/native/native-app-with-c-and-libcurl.md:124]: In this case?

File: docs/topics/native/native-app-with-c-and-libcurl.md [hunk 5]
```diff
@@ -0,0 +1,260 @@
...
+Kotlin/Native helps consume standard C libraries, opening up an entire ecosystem of functionality that exists
+for pretty much anything you may need. Kotlin/Native is already shipped with a set of prebuilt [platform libraries](native-platform-libs.md),
+which provide some additional common functionality to the standard library.
+
+An ideal scenario for interop is to call C functions as if you are calling Kotlin functions, following the same
+signature and conventions. This is when the `cinterop` tool comes in handy. It takes a C library and generates the
+corresponding Kotlin bindings, so that the library can be used as if it were a Kotlin code.
+
+To generate these bindings, let's create a library definition `.def` file that contains some information about
+the necessary headers. In our app, you'll need the `libcurl` library to make some HTTP calls. To create a
```
Comment (koshachy) [inline docs/topics/native/native-app-with-c-and-libcurl.md:105]: In this app

File: docs/topics/native/native-app-with-c-and-libcurl.md [hunk 6]
```diff
@@ -0,0 +1,260 @@
...
+* The second line is the `headerFilter`. It shows what exactly is included. In C, all the headers are also included when
+one file references another one with the `#include` directive. Sometimes it's not necessary, and you can add this
+parameter [using glob patterns](https://en.wikipedia.org/wiki/Glob_(programming)) to fine-tune things.
+
+> `headerFilter` is an optional argument and is mostly used when the library is installed as a system library. You don't
+> want to fetch external dependencies (such as system `stdint.h` header) into the interop library. It may be important to
+> optimize the library size and fix potential conflicts between the system and the provided Kotlin/Native compilation
+> environment.
+>
+{type="note"}
```
Comment (koshachy) [inline docs/topics/native/native-app-with-c-and-libcurl.md:135]: As for me, two notes (one after another) look visually redundant. Can't we rewrite this make a separate para with this text?
Or any other ideas?

File: docs/topics/native/native-app-with-c-and-libcurl.md [hunk 7]
```diff
@@ -0,0 +1,260 @@
...
+> You need to have the `curl` library binaries on your system to make the sample work. On macOS and Linux, it is usually
+> included. On Windows, you can build it from [sources](https://curl.haxx.se/download.html) (you'll need Visual Studio or
+> Windows SDK Commandline tools). For more details, see the [related blog post](https://jonnyzzz.com/blog/2018/10/29/kn-libcurl-windows/).
+> Alternatively, you may also want to consider a [MinGW/MSYS2](https://www.msys2.org/) `curl` binary.
+>
+{type="note"}
+
+## Add interoperability to the build process
+
+To use header files, let's make sure they are generated as a part of the build process. For this, add the following
```
Comment (koshachy) [inline docs/topics/native/native-app-with-c-and-libcurl.md:153]: Omit "let's". This is a tutorial, that contains a list of actions that users should do.

File: docs/topics/native/native-app-with-c-and-libcurl.md [hunk 8]
```diff
@@ -0,0 +1,260 @@
...
+    includeDirs.allHeaders("path")
+}
+```
+
+See the [Interoperability with C](native-c-interop.md) section for more details on the available options.
+
+## Write the application code
+
+Now you have the library and the corresponding Kotlin stubs and can consume them from your application. For this tutorial,
+let's convert the [simple.c](https://curl.haxx.se/libcurl/c/simple.html) example to Kotlin.
```
Comment (koshachy) [inline docs/topics/native/native-app-with-c-and-libcurl.md:208]: let's is redundant

File: docs/topics/native/native-app-with-c-and-libcurl.md [hunk 9]
```diff
@@ -0,0 +1,260 @@
...
+## Compile and run the application
+
+1. Compile the application. To do that, run the following command in the terminal:
+
+```bash
+./gradlew runDebugExecutableNative
+```
+In this case, the `cinterop` generated part is implicitly included in the build.
+
+2. If there are no errors during compilation, run the build by creating a [Run Configuration](https://www.jetbrains.com/help/idea/run-debug-configuration.html#create-permanent).
```
Comment (koshachy) [inline docs/topics/native/native-app-with-c-and-libcurl.md:245]: Maybe some more explanation here, how to run?

Let me explain my idea — I really want readers here to run the app easily, without going to other docs.
And if there is a chance to show here how to create a configuration right for this app, let's provide it.

It's the last step of the tutorial, and I expect that they really want to start it here, without reading any additional info on Run configuration.

File: docs/topics/native/native-app-with-c-and-libcurl.md [hunk 10]
```diff
@@ -0,0 +1,260 @@
...
+IntelliJ IDEA opens the **Run** tab and shows the output — the contents of `https://example.com`:
+
+![Application output with HTML-code](native-output.png){width="700"}
+
+You can see the actual output because the call `curl_easy_perform` prints the result to the standard output. You could
+hide this using `curl_easy_setopt`.
+
+> You can find the resulting code in our [GitHub](https://github.com/Kotlin/kotlin-hands-on-intro-kotlin-native) repository.
+>
```
Comment (koshachy) [inline docs/topics/native/native-app-with-c-and-libcurl.md:254]: TBH, I don't understand the following:
here and there type = "note" line has been omitted.

What was the motivation for this?

**Review by koshachy — CHANGES_REQUESTED** [review]
Some more comments

File: docs/topics/native/native-app-with-c-and-libcurl.md (hunk 10 above)
Comment (danil-pavlov) [reply docs/topics/native/native-app-with-c-and-libcurl.md:253]: To make a different type of note, a tip. This way the article looks less monotonous

**Comment by danil-pavlov** [comment]
@koshachy Ready for the second review, I've implemented most of the suggestions

**Review by koshachy — APPROVED** [review]

**Pull request merged**